src/swtpm/swtpm_nvstore_linear.c

```diff
@@ -0,0 +1,427 @@
+#include "config.h"
+
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <string.h>
+#include <endian.h>
+
+#include <libtpms/tpm_error.h>
+#include <libtpms/tpm_nvfilename.h>
+
+#include "swtpm.h"
+#include "swtpm_debug.h"
+#include "swtpm_nvstore.h"
+#include "swtpm_nvstore_linear.h"
+#include "swtpm_nvstore_linear_file.h"
+#include "logging.h"
+#include "utils.h"
+
+struct nvram_backend_ops nvram_linear_ops = {
+    .prepare = SWTPM_NVRAM_Prepare_Linear,
+    .load    = SWTPM_NVRAM_LoadData_Linear,
+    .store   = SWTPM_NVRAM_StoreData_Linear,
+    .delete  = SWTPM_NVRAM_DeleteName_Linear,
```

Comment:
.cleanup goes here

Reply:
One more general comment. The dir:// backend locks other swtpm instances out from concurrently accessing the directory. I think this one should do that also or does it maybe support multilple swtpm instances accessing a single block device/file concurrently ?